src/dso_api/dynamic_api/filters/values.py

```diff
@@ -0,0 +1,136 @@
+"""Conversion of query-string values to Python types."""
+from __future__ import annotations
+
+import math
+import re
+from datetime import date, datetime, time
+from decimal import Decimal
+
+from django.contrib.gis.geos import GEOSException, GEOSGeometry, Point
+from django.utils.dateparse import parse_datetime
+from django.utils.translation import gettext_lazy as _
+from gisserver.geometries import CRS
+from rest_framework.exceptions import ValidationError
+
+# Don't want Decimal("NaN"), Decimal("-inf") or '0.321000e+2' to be accepted.
+RE_DECIMAL = re.compile(r"^[0-9]+(\.[0-9]+)?$")
+
+
+def str2bool(value: str) -> bool:
+    value = str(value).lower()
+    if value in ("true", "1"):
+        return True
+    elif value in ("false", "0"):
+        return False
+    else:
+        raise ValueError("expect true/false")
+
+
+def str2number(value: str) -> Decimal:
+    if not RE_DECIMAL.match(value):
+        raise ValueError("expecting number")
+
+    return Decimal(value)
```

Comment:
Just let Python do the parsing and check for infinities afterward, it produces better error messages and is probably faster:

```suggestion
    if "." in value:
        value = float(value)
        if not math.isfinite(value):
            raise ValueError(f"not a valid number: {value!r}")
        return value
    return int(value)
```

Reply:
I'm not sure.. I wanted to make sure the input can only be a simple decimal notation, and not `0.321000e+2` for example. That would still work in such case.

Reply:
chosen to leave as it is.